### ISSUE TITLE
`TestOpsGenieGetMessageRecipients` flakiness

### ISSUE THREAD

**Issue by rosstimothy** [gravitational/teleport#40653]
#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/8729348097/job/23951137973

#### Relevant snippet

```
=== RUN   TestOpsGenieGetMessageRecipients/just_approval-schedules
    app_test.go:103: 
        	Error Trace:	/__w/teleport/teleport/integrations/access/accessrequest/app_test.go:103
        	Error:      	Not equal: 
        	            	expected: []common.Recipient{common.Recipient{Name:"foo", ID:"foo", Kind:"", Data:interface {}(nil)}, common.Recipient{Name:"bar", ID:"bar", Kind:"", Data:interface {}(nil)}}
        	            	actual  : []common.Recipient{common.Recipient{Name:"bar", ID:"bar", Kind:"", Data:interface {}(nil)}, common.Recipient{Name:"foo", ID:"foo", Kind:"", Data:interface {}(nil)}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,8 @@
        	            	 ([]common.Recipient) (len=2) {
        	            	+ (common.Recipient) {
        	            	+  Name: (string) (len=3) "bar",
        	            	+  ID: (string) (len=3) "bar",
        	            	+  Kind: (string) "",
        	            	+  Data: (interface {}) <nil>
        	            	+ },
        	            	  (common.Recipient) {
        	            	@@ -6,8 +12,2 @@
        	            	   Data: (interface {}) <nil>
        	            	- },
        	            	- (common.Recipient) {
        	            	-  Name: (string) (len=3) "bar",
        	            	-  ID: (string) (len=3) "bar",
        	            	-  Kind: (string) "",
        	            	-  Data: (interface {}) <nil>
        	            	  }
        	Test:       	TestOpsGenieGetMessageRecipients/just_approval-schedules
```

**Comment by zmb3** [comment]
@EdwardDowling I noticed that this flaky test only exists on `branch/v15` - it is not present in v16 or master. Is that expected?

**Comment by EdwardDowling** [comment]
Yes, that test was to ensure some behavior required for backwards compatibility existed in v15. Thank you for fixing it, did not realize it was flaky.

**Comment by zmb3** [comment]
I didn't fix anything..

**Comment by EdwardDowling** [comment]
Ah sorry i thought that PR was fixing it. Will look at fixing it now

**Comment by zmb3** [comment]
Looks like this was fixed in #40777.
